osu.Game/Beatmaps/BeatmapManager.cs

```diff
@@ -285,24 +282,20 @@ public void Save(BeatmapInfo info, IBeatmap beatmapContent, ISkin beatmapSkin =
         /// <param name="beatmapInfo">The beatmap to lookup.</param>
         /// <param name="previous">The currently loaded <see cref="WorkingBeatmap"/>. Allows for optimisation where elements are shared with the new beatmap. May be returned if beatmapInfo requested matches</param>
         /// <returns>A <see cref="WorkingBeatmap"/> instance correlating to the provided <see cref="BeatmapInfo"/>.</returns>
-        public WorkingBeatmap GetWorkingBeatmap(BeatmapInfo beatmapInfo, WorkingBeatmap previous = null)
+        public virtual WorkingBeatmap GetWorkingBeatmap(BeatmapInfo beatmapInfo, WorkingBeatmap previous = null)
         {
-            if (beatmapInfo?.ID > 0 && previous != null && previous.BeatmapInfo?.ID == beatmapInfo.ID)
-                return previous;
-
-            if (beatmapInfo?.BeatmapSet == null || beatmapInfo == DefaultBeatmap?.BeatmapInfo)
+            if (beatmapInfo?.BeatmapSet == null)
                 return DefaultBeatmap;
 
-            // force a re-query if files are not in a state which looks like the model has
-            // full database information present.
-            if (beatmapInfo.BeatmapSet.Files == null || beatmapInfo.BeatmapSet.Files.Count == 0)
+            // if there are no files, presume the full beatmap info has not yet been fetched from the database.
+            if (beatmapInfo.BeatmapSet.Files.Count == 0)
             {
-                var info = beatmapInfo;
-                beatmapInfo = QueryBeatmap(b => b.ID == info.ID);
+                int lookupId = beatmapInfo.ID;
+                beatmapInfo = QueryBeatmap(b => b.ID == lookupId);
             }
 
-            if (beatmapInfo == null)
-                return DefaultBeatmap;
+            if (beatmapInfo.ID > 0 && previous != null && previous.BeatmapInfo?.ID == beatmapInfo.ID)
+                return previous;
```

Comment:
Does this check really need to happen afterwards? If it's comparing by ID and returning the previous WB, the requery above is basically a no-op as far as I can tell.

Reply:
I think it is indeed redundant, will double check.

Reply:
I actually think this is not required with the `workingCache` logic directly following it, which should cover this exact scenario?

Reply:
Yeah, I don't know why `previous` wasn't removed with the introduction of the cache. Removing this sounds good to me.